orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -520,6 +553,31 @@ public void installFreeIPA(GatewayConfig primaryGateway, List<GatewayConfig> all
         }
     }
 
+    @Override
+    public Optional<String> getFreeIpaMasterHostname(GatewayConfig primaryGateway) throws CloudbreakOrchestratorException {
```

Comment:
why do we have to call salt to find this out? shouldn't be this info available on java side?

Reply:
In a normal situation, the FreeIPA primary gateway is the FreeIPA master. The primary gateway is stored in the FreeIPA database. During a repair of the primary gateway or after a failed repair of the primary gateway, it is likely that the FreeIPA primary gateway is a different node. As far as I can tell, FreeIPA doesn't store the salt roles in the database. I need to ensure that only a single node is identified as the primary FreeIPA instance (if multiple instances generate the CRLs it will be a problem), so I am querying FreeIPA.

If we add the FreeIPA roles into the database, this will be one more thing to synchronize and it will only be used during repair anyway.